sql/moz-fx-data-shared-prod/subscription_platform/stripe_subscriptions/view.sql

```diff
@@ -1,121 +1,151 @@
-    ) AS customer_start_date,
+    _fivetran_synced,
+    -- TODO verify logic for subscription_start_date
     COALESCE(trial_end, start_date) AS subscription_start_date,
     created,
+    -- trial_start,
     trial_end,
+    -- current_period_end,
+    -- current_period_start,
     canceled_at,
-    mozfun.map.get_key(metadata, "cancelled_for_customer_at") AS canceled_for_customer_at,
+    JSON_VALUE(metadata, "$.cancelled_for_customer_at") AS canceled_for_customer_at,
     cancel_at,
     cancel_at_period_end,
-    ended_at,
-    discount.promotion_code AS promotion_code_id,
+    IF(ended_at <= _fivetran_synced, ended_at, NULL) AS ended_at,
   FROM
-    mozdata.stripe.subscriptions
+    `moz-fx-data-bq-fivetran`.stripe.subscription_history
   WHERE
     status NOT IN ("incomplete", "incomplete_expired")
+    -- choose subscription from subscription history
+    AND _fivetran_active
 ),
-stripe_customers AS (
+subscription_item AS (
+  SELECT
+    id AS subscription_item_id,
+    subscription_id,
+    plan_id,
+  FROM
+    `moz-fx-data-bq-fivetran`.stripe.subscription_item
+),
+customer AS (
   SELECT
     id AS customer_id,
-    mozfun.map.get_key(metadata, "fxa_uid") AS fxa_uid,
+    COALESCE(
+      TO_HEX(SHA256(JSON_VALUE(customer.metadata, "$.userid"))),
+      JSON_VALUE(pre_fivetran_customer.metadata, "$.fxa_uid")
+    ) AS fxa_uid,
+    COALESCE(customer.address_country, pre_fivetran_customer.address_country) AS address_country,
   FROM
-    mozdata.stripe.customers
+    `moz-fx-data-bq-fivetran`.stripe.customer
+  FULL JOIN
+    -- include customers that were deleted before the initial fivetran stripe import
+    `moz-fx-data-shared-prod`.stripe_external.pre_fivetran_customer
+  USING
+    (id)
 ),
-stripe_charges AS (
+charge AS (
   SELECT
-    id AS charge_id,
-    payment_method_details.card.country,
+    charge.id AS charge_id,
+    COALESCE(card.country, charge.billing_detail_address_country) AS country,
   FROM
-    `moz-fx-data-shared-prod`.stripe_external.charges_v1
+    `moz-fx-data-bq-fivetran`.stripe.charge
+  JOIN
+    `moz-fx-data-bq-fivetran`.stripe.card
+  ON
+    charge.card_id = card.id
   WHERE
-    status = "succeeded"
+    charge.status = "succeeded"
 ),
-stripe_invoice_lines AS (
+invoice_provider_country AS (
   SELECT
-    lines.subscription AS subscription_id,
+    invoice_line_item.subscription_id,
     IF(
-      "paypalTransactionId" IN (SELECT key FROM UNNEST(invoices_v1.metadata)),
-      -- FxA copies paypal billing agreement location to customer_address
-      STRUCT("Paypal" AS provider, invoices_v1.customer_address.country),
-      ("Stripe", stripe_charges.country)
+      JSON_VALUE(invoice.metadata, "$.paypalTransactionId") IS NOT NULL,
+      -- FxA copies paypal billing agreement country to customer address
+      STRUCT("Paypal" AS provider, customer.address_country AS country),
+      ("Stripe", charge.country)
     ).*,
-    invoices_v1.event_timestamp,
+    invoice.finalized_at,
```

Comment:
`finalized_at` may be null, so using `created` might be preferable.

Reply:
okay